etcd-manager/pkg/etcd/etcdserver.go

```diff
@@ -604,12 +605,22 @@ func (s *EtcdServer) startEtcdProcess(state *protoetcd.EtcdState) error {
 		return err
 	}
 
-	binDir, err := BindirForEtcdVersion(state.EtcdVersion, "etcd")
```

Comment:
So I'm sorry, but I'm still worried about this... 

This particular method (startEtcdProcess) is called from multiple places, and I think for some of them we should use `EtcdVersionForAdoption`, and for some of them we should use the version the caller asked for.

I also think we should report our version accurately in calls to GetInfoRequest, so we should populate state.EtcdVersion with whatever version we are going to run.

I'm digging into the details of these callsites though - you're usually right on these things, but I need to make sure :-)